diag_manager/fms_diag_field_object.F90

```diff
@@ -426,10 +441,30 @@ function get_send_data_time(this) &
   rslt = this%input_data_buffer%get_send_data_time()
 end function get_send_data_time
 
+!> @brief Prepare the input_data_buffer to do the reduction method
+subroutine prepare_data_buffer(this)
+  class (fmsDiagField_type) , intent(inout):: this                !< The field object
+
+  if (.not. this%multiple_send_data) return
+  if (this%mask_variant) return
```

Comment:
looks like these could be combined

Reply:
Why not wrap this function with these IF statements?

Reply:
I think it is cleaner like this.